packages/compass-e2e-tests/helpers/commands/listen-for-telemetry-events.ts

```diff
@@ -6,6 +6,7 @@ export async function listenForTelemetryEvents(
   browser: CompassBrowser,
   telemetry: Telemetry
 ): Promise<(eventName: string) => Promise<any>> {
+  await telemetry.pollForEvents?.(browser);
```

Comment:
For compass-web tracking is stored on the window object and copying it out is async. I thought about making telemetry.events() async, but it needs the browser object and we don't have that everywhere we use telemetry anyway. (There are some tests that test the compass CLI, for example.)

Not an issue for the usual compass telemetry because there it arrives in the telemetry server in the background while this code polls events().

I did add a check to startFakeTelemetry that should catch most cases where we try and use this without adding polling at least.